pkg/services/sqlstore/transactions_test.go

```diff
@@ -65,3 +68,49 @@ func TestIntegrationReuseSessionWithTransaction(t *testing.T) {
 		}))
 	})
 }
+
+func TestIntegrationPublishAfterCommitWithNestedTransactions(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ss := InitTestDB(t)
+	ctx := context.Background()
+
+	// On X success
+	var xHasSucceeded bool
+	ss.Bus().AddEventListener(func(ctx context.Context, e *X) error {
+		xHasSucceeded = true
+		fmt.Printf("Succeeded and committed: %T\n", e)
```

Comment:
Please use [`t.Logf`](https://pkg.go.dev/testing#T.Logf) throughout the test.

> Logf formats its arguments according to the format, analogous to Printf, and records the text in the error log. A final newline is added if not provided. For tests, the text will be printed only if the test fails or the -test.v flag is set. For benchmarks, the text is always printed to avoid having performance depend on the value of the -test.v flag.